engines/tot/saveload.cpp

```diff
@@ -0,0 +1,291 @@
+		delete (g_engine->_conversationData);
+		byte *convBuf = (byte *)malloc(size);
+		s.syncBytes(convBuf, size);
+		debug("Loading conversation data now");
+		g_engine->_conversationData = new Common::MemorySeekableReadWriteStream(convBuf, size, DisposeAfterUse::NO);
+	}
+	return true;
+}
+
+bool syncItemData(Common::Serializer &s, Common::MemorySeekableReadWriteStream *sceneObjects) {
+	int size = sceneObjects->size();
+	if (s.isSaving()) {
+		byte *objBuf = (byte *)malloc(size);
+		sceneObjects->seek(0, 0);
+		sceneObjects->read(objBuf, size);
+		s.syncBytes(objBuf, size);
+		free(objBuf);
+	}
+	if (s.isLoading()) {
+		delete (g_engine->_sceneObjectsData);
+		byte *objBuf = (byte *)malloc(size);
+		s.syncBytes(objBuf, size);
+		debug("Loading item data now");
+		g_engine->_sceneObjectsData = new Common::MemorySeekableReadWriteStream(objBuf, size, DisposeAfterUse::NO);
+	}
+	return true;
+}
+
+Common::Error syncSaveData(Common::Serializer &ser, SavedGame &game) {
+	if (!syncGeneralData(ser, game)) {
+		warning("Error while synchronizing general data");
+		return Common::kUnknownError;
+	}
+	if (!syncRoomData(ser, g_engine->_rooms)) {
+		warning("Error while synchronizing room data");
+		return Common::kUnknownError;
+	}
+	if (!syncItemData(ser, g_engine->_sceneObjectsData)) {
+		warning("Error while syncrhonizing object data");
+		return Common::kUnknownError;
+	}
+	if (!syncConversationData(ser, g_engine->_conversationData)) {
+		warning("Error while syncrhonizing conversation data");
+		return Common::kUnknownError;
+	}
+	if (ser.err()) {
+		warning("Error while synchronizing");
+		return Common::kUnknownError;
+	}
+
+	return Common::kNoError;
+}
+
+Common::Error TotEngine::syncGame(Common::Serializer &s) {
+	Common::Error result;
+
+	if (s.isLoading()) {
+		debug("Loading game!!");
+		SavedGame loadedGame;
+		// Means we are loading from before the game has started
+		// if(rooms == nullptr) {
+		g_engine->_graphics->clear();
+		displayLoading();
+
+		loadCharAnimation();
+		loadInventory();
+
+		g_engine->_graphics->loadPaletteFromFile("DEFAULT");
+		initScreenPointers();
+
+		g_engine->_graphics->totalFadeOut(0);
+		g_engine->_graphics->clear();
+		initializeScreenFile();
+		initializeObjectFile();
+		readConversationFile();
+
+		result = syncSaveData(s, loadedGame);
+		loadGame(loadedGame);
+	} else {
+		saveGameToRegister();
+		result = syncSaveData(s, _savedGame);
+	}
+	return result;
+}
+
+Common::Error TotEngine::saveGameStream(Common::WriteStream *stream, bool isAutosave) {
+	const byte version = SAVEGAME_CURRENT_VERSION;
+	Common::Serializer s(nullptr, stream);
+	s.setVersion(version);
+	stream->writeByte(version);
+
+	return syncGame(s);
+}
+Common::Error TotEngine::loadGameStream(Common::SeekableReadStream *stream) {
+	byte version = stream->readByte();
+	if (version > SAVEGAME_CURRENT_VERSION) {
+		GUI::MessageDialog dialog(Common::String("Saved game was created with a newer version of ScummVM. Unable to load."));
```

Comment:
```suggestion
		GUI::MessageDialog dialog(_("Saved game was created with a newer version of ScummVM. Unable to load."));
```
That is, mark it as translatable, and add this file to POTFILES. Don't forget `#include "common/translations.h"